### PR TITLE
 Support OmniGibson and Isaac Sim installation via pip

### PR DESCRIPTION
**Pull request by yyf20001230**
Changes:

* Add an optional dependency on our Isaac Sim metapackage
* Add compatibility for launcher-installed or pip-installed SimulationApp in simulator.py
* Update kit file to work with both pip-installed and launcher-installed Isaac Sim
* Move dataset downloader so it gets shipped on PyPI too

### CODE REVIEW

File: scripts/setup.sh
Comment (cgokmen) [inline scripts/setup.sh:17]: Undo all changes to this file.

File: tests/test_multiple_envs.py
Comment (cgokmen) [inline tests/test_multiple_envs.py:122]: Undo

File: setup.py
Comment (cgokmen) [inline setup.py:50]: Remove the 2nd line

File: omnigibson/simulator.py
Comment (cgokmen) [inline omnigibson/simulator.py:157]: Undo this

File: omnigibson/simulator.py
Comment (cgokmen) [inline omnigibson/simulator.py:114]:
```
if "ISAAC_PATH" in os.environ:
    ...
else:
    ...
```

File: omnigibson/simulator.py
Comment (cgokmen) [inline omnigibson/simulator.py:143]: if "EXP_PATH" in os.environ:
    ...
else:
    ...

Comment (cgokmen) [reply omnigibson/simulator.py:143]: Find out a more principled way for getting the isaacsim package's path (e.g. you can try importing isaacsim and then doing isaacsim.__package__ or maybe isaacsim.__file__?)

import isaacsim
exp_path = os.path.join(os.path.dirname(isaacsim.__file__), "isaacsim", "apps", kit_file_name)

File: scripts/setup.sh
Comment (cgokmen) [inline scripts/setup.sh:68]: Undo this change

**Review by cgokmen — APPROVED** [review]
Please wait for a review from Josiah too prior to merging. Thanks for the great work.

File: docs/getting_started/installation.md
Comment (cremebrule) [inline docs/getting_started/installation.md:47]: this is awesome!!

File: docs/tutorials/running_on_a_server.md
Comment (cremebrule) [inline docs/tutorials/running_on_a_server.md:20]: maybe this ID needs to be updated? @cgokmen

File: omnigibson/omnigibson_4_0_0.kit
Comment (cremebrule) [inline omnigibson/omnigibson_4_0_0.kit:25]: what's the reason for not using these imports?

File: omnigibson/simulator.py
Comment (cremebrule) [inline omnigibson/simulator.py:115]: nit: combine into single line? `isaac_path = os.environ["ISAAC_PATH"] if isaacsim is None else os.path.dirname(isaacsim.__file__)`

File: omnigibson/simulator.py
Comment (cremebrule) [inline omnigibson/simulator.py:110]: interesting -- so when pip installing, are the LD_LIBRARY_PATH / PYTHONPATH env variables auto-updated when importing isaacsim? Formerly we had to sort of a do a hacky thing where every time you activated the conda env to run omnigibson we would automatically run the setup script from isaac so that these paths were updated.

**Review by cremebrule — COMMENTED** [review]
overall LGTM! Just a couple comments

File: omnigibson/simulator.py
Comment (cgokmen) [reply omnigibson/simulator.py:110]: Yes, I believe they get updated upon import. Pretty neat.

File: omnigibson/omnigibson_4_0_0.kit
Comment (cgokmen) [reply omnigibson/omnigibson_4_0_0.kit:25]: They are actually excluded in the kit file shipped in pip-installed Isaac Sim, both in the kit file and in the actual exts directory, meaning if we keep these OG won't work with the pip installed version (those extensions cant be found). But these I believe are some off-the-shelf models etc. that we don't use, looks very safe to remove.

File: docs/tutorials/running_on_a_server.md
Comment (cgokmen) [reply docs/tutorials/running_on_a_server.md:20]: I am updating these but I think it's safe to let them fall behind later. People should hopefully get the gist.

File: omnigibson/simulator.py
Comment (cgokmen) [reply omnigibson/simulator.py:115]: Done

**Pull request merged**
